File: frontend/src/types/graph.ts
export type EntityType =
  | "Person"
  | "Band"
  | "Work"
  | "Album"
  | "Genre"
  | "Location"
  | "Event"
  | "Title";

export interface SearchItem {
  id: string;
  name: string;
  type: string;
  summary?: string | null;
}

export interface EntityDetails {
  id: string;
  name: string;
  type: string;
  aliases: string[];
  summary?: string | null;
  meta: Record<string, unknown>;
}

export interface GraphNode {
  id: string;
  label: string;
  type: string;
  summary?: string | null;
  meta: Record<string, unknown>;
}

export interface GraphLink {
  id: string;
  source: string;
  target: string;
  type: string;
  meta: Record<string, unknown>;
}

export interface GraphData {
  nodes: GraphNode[];
  links: GraphLink[];
}

export interface PathResponse extends GraphData {
  source_id: string;
  target_id: string;
}

export interface GraphRagEvidence {
  type: string;
  text: string;
}

export interface GraphRagResponse {
  question: string;
  answer: string;
  mode: string;
  matched_entities: SearchItem[];
  evidence: GraphRagEvidence[];
  graph: GraphData;
}

export interface HealthResponse {
  status: string;
  mode: string;
}

File: backend/app/schemas/graph.py
from pydantic import BaseModel, Field


class SearchItem(BaseModel):
    id: str
    name: str
    type: str
    summary: str | None = None


class SearchResponse(BaseModel):
    items: list[SearchItem]


class EntityDetails(BaseModel):
    id: str
    name: str
    type: str
    aliases: list[str] = Field(default_factory=list)
    summary: str | None = None
    meta: dict[str, object] = Field(default_factory=dict)


class GraphNode(BaseModel):
    id: str
    label: str
    type: str
    summary: str | None = None
    meta: dict[str, object] = Field(default_factory=dict)


class GraphLink(BaseModel):
    id: str
    source: str
    target: str
    type: str
    meta: dict[str, object] = Field(default_factory=dict)


class GraphResponse(BaseModel):
    nodes: list[GraphNode] = Field(default_factory=list)
    links: list[GraphLink] = Field(default_factory=list)


class PathResponse(GraphResponse):
    source_id: str
    target_id: str


class HealthResponse(BaseModel):
    status: str
    mode: str
